api/v1alpha1/frontend_types.go

```diff
@@ -115,6 +115,7 @@ type FrontendSpec struct {
 	ServiceTiles []*ServiceTile `json:"serviceTiles,omitempty" yaml:"serviceTiles,omitempty"`
 	// Data for the available widgets for the resource
 	WidgetRegistry []*WidgetEntry `json:"widgetRegistry,omitempty" yaml:"widgetRegistry,omitempty"`
+	Replicas       *int32         `json:"replicas,omitempty" yaml:"replicas,omitempty"`
```

Comment:
Just out of curiosity. Does this need to be a pointer? I know the check would have to change bellow but I am curious why we prefer pointer here over plain int value.

Reply:
Well, I could be wrong so it isn't a hill I would die on, but as I understood it if the type in the spec is int, then if the property is omitted from the resource then the value will be implicitly set to 0, which would then make my comparison to 0. 0 is a valid value for replicas, and can even be useful in some situations for scaling down and then up again. A corner case, but it has happened. More importantly, it would mean that my logic would be"if the value is 0, set the value to 1" which violates the principle of least surprise. So, I thought this was more correct. That said if any part of my understanding of how all of this hangs together is wrong I'm happy to change it.

Reply:
I see. Ok, I was not aware that 0 is a valid replica value. I get it now.